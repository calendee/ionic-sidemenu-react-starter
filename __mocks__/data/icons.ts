import {
  americanFootball,
  basketball,
  beer,
  bluetooth,
  boat,
  build,
  flask,
  football,
  paperPlane,
  wifi,
} from 'ionicons/icons';

import { IIcon } from '../../src/declarations';

export const icons: IIcon[] = [
  { icon: flask, name: 'flask', id: 1 },
  { icon: wifi, name: 'wifi', id: 2 },
  { icon: beer, name: 'beer', id: 3 },
  { icon: football, name: 'football', id: 4 },
  { icon: basketball, name: 'basketball', id: 5 },
  { icon: paperPlane, name: 'paperPlane', id: 6 },
  { icon: americanFootball, name: 'americanFootball', id: 7 },
  { icon: boat, name: 'boat', id: 8 },
  { icon: bluetooth, name: 'bluetooth', id: 9 },
  { icon: build, name: 'build', id: 10 },
];
