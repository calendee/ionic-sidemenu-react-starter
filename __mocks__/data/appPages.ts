import { AppPage } from '../../src/declarations';
import { home, list } from 'ionicons/icons';

const appPages: AppPage[] = [
  {
    title: 'Menu Link To Home',
    url: '/home',
    icon: home,
  },
  {
    title: 'Menu Link To List',
    url: '/home/list',
    icon: list,
  },
];

export default appPages;
