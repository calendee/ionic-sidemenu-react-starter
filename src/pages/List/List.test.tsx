import React from 'react';

import ListPage, { ListItems } from './List';
import { render } from '../../../test/test-utils';
import { icons } from '../../../__mocks__/data/icons';

describe('List Page', () => {
  test('renders', () => {
    const { container, getByText } = render(<ListPage />);
    getByText(/list/i);
    expect(container).toMatchSnapshot();
  });

  test('renders list items', () => {
    const { getByText } = render(<ListItems icons={icons} />);
    icons.forEach(({ icon, name, id }) => {
      const regex = new RegExp(`^this is item # ${id}$`, 'i');
      getByText(regex);
    });
  });
});
