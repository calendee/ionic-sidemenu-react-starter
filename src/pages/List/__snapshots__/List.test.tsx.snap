// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Page renders 1`] = `
<div>
  <div
    class="ion-page"
    data-testid="list-page"
  >
    <ion-header>
      <ion-toolbar>
        <ion-buttons>
          <ion-menu-button />
        </ion-buttons>
        <ion-title>
          List
        </ion-title>
      </ion-toolbar>
    </ion-header>
    <ion-content>
      <ion-list>
        <ion-item>
          <ion-icon />
          Item 
          1
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            1
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          2
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            2
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          3
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            3
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          4
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            4
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          5
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            5
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          6
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            6
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          7
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            7
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          8
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            8
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          9
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            9
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          10
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            10
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          11
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            11
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          12
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            12
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          13
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            13
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          14
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            14
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          15
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            15
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          16
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            16
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          17
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            17
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          18
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            18
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          19
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            19
          </div>
        </ion-item>
        <ion-item>
          <ion-icon />
          Item 
          20
          <div
            class="item-note"
            slot="end"
          >
            This is item # 
            20
          </div>
        </ion-item>
      </ion-list>
    </ion-content>
  </div>
</div>
`;
